### ISSUE TITLE
Running tests on an APK instrumented with older version of SDK returns Altom.AltDriver.NullReferenceException

### ISSUE THREAD

**Issue by AleVescan** [alttester/AltTester-Unity-SDK#1254]
**In what language is the test written?** 
(C#, Java, Python 2.7, Python 3)
C#

**What version of AltTester Unity SDK are you using?**
AltTester Unity SDK package used in tests : 1.8.*
AltTester Unity SDK used to instrument and generate apk : 1.7.*

**On what device are you trying to run the test?**
Android

**The OS from where you are running the test?**
Windows

### Summary

When the user tries to run tests on Android, using an APK instrumented with an older version of SDK e.g. v.1.7.0 the tests fail, returning the following error: **Altom.AltDriver.NullReferenceException : Object reference not set to an instance of an object**. 
Trying to connect the same application to AltTester Desktop returns the following message : **IncompatibleVersionException: Version mismatch. Please instrument your game with AltTester Unity SDK v.1.8.*. You are currently using AltTester Unity SDK v. 1.7.0.**

### Steps to reproduce

1. Install an application that was instrumented with an older version of SDK, e.g. APK retrieved when cloning the repository, not when downloading the app from the link. 
https://github.com/alttester/EXAMPLES-CSharp-Android-AltTrahCat/tree/master/app
2.  Make sure the device has USB debugging on. 
3. Set port forwarding adb forward tcp:13000 tcp:13000
4. Start the application and run some C# Tests . E.g the tests from the same repository https://github.com/alttester/EXAMPLES-CSharp-Android-AltTrahCat
5. Remove port forwarding adb forward --remove-all
6. Open AltTester Desktop and connect to the application 

### Result 

After step 4 most of the tests fail due to 
Error Message:
   **Altom.AltDriver.NullReferenceException : Object reference not set to an instance of an object**

Checking AltTester Desktop logs after step 6 the following exception is thrown **IncompatibleVersionException: Version mismatch. Please instrument your game with AltTester Unity SDK v.1.8.*. You are currently using AltTester Unity SDK v. 1.7.0.**

### Expected result

A more informative error from Altom.AltDriver, similar to the one returned by AltTesterDesktop. In that way the user can fix the issue on their part easier.  

### Notes

Following the same steps with an APK instrumented with a higher SDK version works without issues. 
An updated APK can be found here : https://altom.com/app/uploads/AltTester/TrashCat/TrashCatAndroid.zip

### Relevant logs and/or screenshots
![Screenshot 2023-05-11 102034](https://github.com/alttester/AltTester-Unity-SDK/assets/126661357/b1d896ff-9b7f-466f-bbd4-0be078b57d80)
![Screenshot 2023-05-11 102608](https://github.com/alttester/AltTester-Unity-SDK/assets/126661357/aadf9b07-58bf-47b4-9b2a-663222ec0d5b)

**Comment by sandraaltom** [comment]
Triaged Week #38, 2023. 
We won't fix this, as this is the expected behaviour.
